evaluator/builtin_string.go

```diff
@@ -186,6 +187,26 @@ func builtinReverse(args []types.Datum, _ context.Context) (d types.Datum, err e
 	}
 }
 
+// See: http://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_space
+func builtinSpace(args []types.Datum, _ context.Context) (d types.Datum, err error) {
+	x := args[0]
+	switch x.Kind() {
```

Comment:
if x.IsNull() {
    return d, nil
}
Then you do not need the long default part.